### DISCUSSION TITLE
YOLOAir 🌟 Supported Updates

### DISCUSSION THREAD

**Discussion by iscyy** [iscyy/ultralyticsPro#43] [Ideas]
已经完成更新 20+ 种 Transformer系列及其变种主干网络🌟、多种MLP网络🌟 以及 大部分重参数化结构模型网络 🚀🚀🚀
已经完成更新 图神经网络🌟 在YOLO中的应用
已经完成更新 多模态🔥 在YOLO中的应用
已经完成更新 30+ 种 Attention注意力机制🌟、
已经完成更新 多种Head检测头
完成更新支持YOLOv6模型-v2.0 paper版本
已经完成更新 YOLO系列网络模型热力图可视化 (GardCAM、GardCAM++等)支持YOLOv3、YOLOv3-SPP、YOLOv4、YOLOv5、YOLOR、YOLOv7、Scaled_YOLOv4、TPH-YOLO、SPD-YOLO、PP-YOLO以及自定义网络模型等模型
...

**Comment by iscyy** [comment]
Beta version update: Added support for YOLOv6-v2.0 (latest Paper) model

**Comment by iscyy** [comment]
<img width="518" alt="yoloair" src="https://user-images.githubusercontent.com/24501320/189464307-c216ee8f-ff1a-4e93-90cb-8bf5bf79e0f8.png">
✅完成 将yolov6-v2.0最新paper版本框架代码更新到 yoloair 框架代码中，包括网络结构和不同Loss策略配置,训练方式 等更新

**Comment by iscyy** [comment]
The yolov6 implementation of yoloair, part of the training log is shown, Include label assignment policy
<img width="736" alt="yoloair" src="https://user-images.githubusercontent.com/24501320/189467427-078056de-eb43-40a9-af87-f3140e714fcf.png">

**Comment by iscyy** [comment]
Beta🚀
✅Completed the refactoring and updating of the PPYOLOE network model code to the YOLOAir framework code, including Anchor free, CSPRepResStage, head mechanism ET-head using Varifocal Loss (VFL) and Distribution focal loss (DFL), and dynamic label assignment algorithm Task Alignment Learning (TAL)

**Comment by iscyy** [comment]
<img width="535" alt="yoloair" src="https://user-images.githubusercontent.com/24501320/189507909-36cb80f4-95cd-4cc1-a2ad-161a69379559.png">

**Comment by iscyy** [comment]
Beta✅YOLOAir Integrate Task-aligned One-stage Object Detection (TOOD) to explicitly align two tasks in a learning-based manner

**Comment by iscyy** [comment]
✅ Integrated label assignment strategy "A Dual Weighting Label Assignment Scheme for Object Detection"

**Comment by iscyy** [comment]
- 完成更新支持 **20+ 种 Transformer系列及其变种主干网络**🌟、**多种MLP网络🌟** 以及 **大部分重参数化结构模型网络** 
- 完成更新支持 **图神经网络**🌟 在YOLO中的应用🌟
- 完成更新支持 多模态🔥 在YOLO中的应用🌟
- 完成更新支持 30+ 种 **Attention注意力机制**🌟
- 完成更新支持 YOLOv6模型-v2.0 paper 版本🌟
- 完成更新支持 **YOLO系列网络模型热力图可视化**🌟
(GardCAM、GardCAM++等)支持YOLOv3、YOLOv3-SPP、YOLOv4、YOLOv5、YOLOR、YOLOv7、Scaled_YOLOv4、TPH-YOLO、SPD-YOLO、PP-YOLO以及自定义网络模型等模型🌟
- 完成更新集成 TOOD 算法🌟
- 集成支持 Adaptive Training Sample Selection 标签分配策略🌟
- 集成支持 Task Alignment Learning 标签分配策略🌟
- 完成更新集成 PPYOLOE 算法模型🌟
- 完成更新集成 PPYOLOE Plus 算法模型🌟  
News:
- 集成支持 AutoAssign Label Assignment🌟
- 集成支持 Dual Weighting Label Assignment🌟
- 集成支持 FreeAnchor 算法🌟
- 集成支持 多种 Head 检测头 🌟

**Reply by Seperendity** [reply]
您好，请问图神经和多模态的内容在项目的哪部分呢，没有找到哈，想学习一下

**Comment by iscyy** [comment]
<li>RepMLP系列</li>
        <li>ACNet系列</li>
        <li>RepConv系列</li>
        <li>OREPA系列</li>
        <li>Mobileone系列</li>
        <li>自注意力Transformer系列</li>
        <li>MobileViT系列</li>
        <li>BoTNet-Transfomrer</li>
        <li>CoTNet-Transfomrer</li>
        <li>Swin-Transfomrer</li>
        <li>Swin-Transfomrer(v2)</li>
        <li>CMT Transformer</li>
        <li>PvT Transformer</li>
        <li>PvT Transformer(v2)</li>
        <li>XCiT Transformer</li>
        <li>LVViT Transformer</li>
        <li>Neighborhood-Attention-Transformer</li>

**Comment by iscyy** [comment]
<li>YOLOv4Head</li>
          <li>YOLOv5Head</li>
          <li>YOLOv6 Efficient decoupled head </li>
          <li>YOLOv7 IDetect Head</li>
          <li>YOLOv7 IAuxDetect Head</li>
          <li>PPYOLOE Efficient Task-aligned head with DFL and VFL</li>
          <li>YOLOX DetectX Head</li>
          <li>TOODHead</li>
          <li>自适应空间特征融合 检测头ASFF Head</li>
          <li>YOLOR 隐式学习 Head</li>
          <li>EH-Head</li>
          <li>Dual Weighting Head</li>
          <li>FCOS Head</li>
          <li>Dynamic Head</li>

**Comment by iscyy** [comment]
- Multi Anchor策略
- YOLOv5 标签分配策略
- SimOTA 标签分配策略
- YOLOv7 标签分配策略
- Adaptive Training Sample Selection 标签分配策略
- Task Alignment Learning 标签分配策略
- AutoAssign 标签分配策略🌟
- Dual Weighting 标签分配策略🌟
- FreeAnchor 标签分配策略🌟

**Comment by iscyy** [comment]
Beta内测版✅已更新 YOLOAir / YOLOAir2 集成 `DAMOYOLO`模型的 改进`AlignOTA`标签分配策略
